examples/http.rs

```diff
@@ -0,0 +1,65 @@
+extern crate may;
+extern crate httparse;
+extern crate bytes;
+
+use may::coroutine;
+use may::net::TcpListener;
+use httparse::Request;
+use httparse::Status;
+use bytes::BufMut;
+use std::io::{Read, Write};
+
+fn req_done(buf: &[u8], path: &mut String) -> Option<usize> {
+    let mut headers = [httparse::EMPTY_HEADER; 16];
+    let mut req = httparse::Request::new(&mut headers);
+
+    if let Ok(Status::Complete(i)) = req.parse(buf) {
+        path.clear();
+        path.push_str(req.path.unwrap_or("/"));
+        return Some(i);
+    }
+
+    None
+}
+
+fn main() {
+    may::config().set_io_workers(4);
+
+    let listener = TcpListener::bind("127.0.0.1:8080").unwrap();
+    while let Ok((mut stream, _)) = listener.accept() {
+        coroutine::spawn(move || {
+            let mut buf = Vec::new();
+            let mut path = String::new();
+
+            loop {
+                if let Some(i) = req_done(&buf, &mut path) {
+                    let response = match &*path {
+                        "/" => "Welcome to May http demo\n",
+                        "/hello" => "Hello, World!\n",
+                        "/quit" => std::process::exit(1),
+                        _ => "Cannot find page\n"
+                    };
+                   
+                    write!(stream, "\
+                    HTTP/1.1 200 OK\r\n\
+                    Server: May\r\n\
+                    Content-Length: {}\r\n\
+                    Date: 1-1-2000\r\n\
+                    \r\n\
+                    {}", response.len(), response);
```

Comment:
directly call the `write!` macro on a raw writer would be very slow. I think internally it calls too much system write. change the code as following gives me a shock about wrk bench result
```rust
                    let s = format!("\
                        HTTP/1.1 200 OK\r\n\
                        Server: May\r\n\
                        Content-Length: {}\r\n\
                        Date: 1-1-2000\r\n\
                        \r\n\
                        {}", response.len(), response);

                    stream.write_all(s.as_bytes());
```

Reply:
by the way, you can use `std::io::BufWriter`  for internal cache things.